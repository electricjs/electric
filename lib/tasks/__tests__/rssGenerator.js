'use strict';

const fs = require('fs-extra');
const gulp = require('gulp');
const path = require('path');

const runSequence = require('run-sequence').use(gulp);

const registerTasks = require('../../index').registerTasks;
const sitePath = path.join(
	__dirname,
	'../../../test/fixtures/sites/rss-generator-site'
);

beforeAll(() => {
	process.chdir(sitePath);

	registerTasks({
		gulp: gulp
	});
});

afterAll(done => {
	runSequence('clean', function() {
		done();
	});
});

test('it should generate glob.rss file',
	done => {
		runSequence('front-matter', function() {
			const config = {encoding: 'utf8'};
			const rssFeed = fs.readFileSync(
				path.join(sitePath, 'dist/blog/rss/blog.xml'),
				config
			);
			const rssArr = rssFeed.split('\n');

			// Removes timestamp so snapshot passes
			rssArr.splice(11, 1);

			expect(rssArr.join('\n')).toMatchSnapshot();

			done();
		});
	}
);
