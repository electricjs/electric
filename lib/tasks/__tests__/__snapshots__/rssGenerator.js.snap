// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should generate glob.rss file 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><rss xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\" version=\\"2.0\\" xmlns:geo=\\"http://www.w3.org/2003/01/geo/wgs84_pos#\\">
    <channel>
        <title><![CDATA[rss-generator-site]]></title>
        <description><![CDATA[This is the description of your blogs RSS feed]]></description>
        <link>https://027ffbd9.ngrok.io</link>
        <image>
            <url>example.png</url>
            <title>rss-generator-site</title>
            <link>https://027ffbd9.ngrok.io</link>
        </image>
        <generator>RSS for Node</generator>
        <atom:link href=\\"https://027ffbd9.ngrok.io/blog/rss/blog.xml\\" rel=\\"self\\" type=\\"application/rss+xml\\"/>
        <copyright><![CDATA[2017]]></copyright>
        <language><![CDATA[English]]></language>
        <managingEditor><![CDATA[Name of person who manages]]></managingEditor>
        <webMaster><![CDATA[Name of person who manages]]></webMaster>
        <ttl>60</ttl>
        <item>
            <title><![CDATA[Markdown Post]]></title>
            <description><![CDATA[This blog is super duper cool.]]></description>
            <link>https://027ffbd9.ngrok.io/blog/markdown-post.html</link>
            <guid isPermaLink=\\"true\\">https://027ffbd9.ngrok.io/blog/markdown-post.html</guid>
            <category><![CDATA[cats]]></category>
            <category><![CDATA[dogs]]></category>
            <dc:creator><![CDATA[Ryan Garant]]></dc:creator>
            <pubDate>Thu, 02 Feb 2017 08:00:00 GMT</pubDate>
            <geo:lat>51.503364</geo:lat>
            <geo:long>-0.127625</geo:long>
            <enclosure url=\\"https://027ffbd9.ngrok.io/images/example.png\\" length=\\"0\\" type=\\"image/png\\"/>
        </item>
        <item>
            <title><![CDATA[Soy Post]]></title>
            <description><![CDATA[This soy post was made from electric boilerplates.]]></description>
            <link>https://027ffbd9.ngrok.io/blog/soy-post.html</link>
            <guid isPermaLink=\\"true\\">https://027ffbd9.ngrok.io/blog/soy-post.html</guid>
            <dc:creator><![CDATA[Joe Doe]]></dc:creator>
            <pubDate>Thu, 12 Jan 2017 08:00:00 GMT</pubDate>
        </item>
    </channel>
</rss>"
`;
